app/scripts/modules/core/src/account/account.service.ts

```diff
@@ -152,29 +147,11 @@ export class AccountService {
       });
   }
 
-  public getUniqueGceZonesForAllAccounts(provider: string): ng.IPromise<IZone> {
-    return this.getCredentialsKeyedByAccount(provider)
-      .then((regions: IAggregatedAccounts) => {
-        return chain(regions)
-          .map('regions')
-          .flatten()
-          .reduce((zone: IZone, object: IZone) => {
-            Object.keys(object).forEach((key: string) => {
-              if (zone[key]) {
-                zone[key] = Array.from(new Set([...zone[key], ...object[key]]));
-              } else {
-                zone[key] = object[key];
-              }
-            });
-
-            return zone;
-          }, {})
-          .value();
-      });
-  }
-
-  public listAccounts(provider: string = null, providerVersion: string = null): ng.IPromise<IAccount[]> {
-    let result: ng.IPromise<IAccount[]> = this.API.one('credentials').useCache().get();
+  public listAccounts(provider: string = null, providerVersion: string = null): IPromise<IAccountDetails[]> {
+    if (!this.accountCache) {
```

Comment:
why did you do this manual caching instead of the `.API.useCache()` thing?

Reply:
it's dumb, I wanted to avoid the extra serializing/deserializing from strings to json, I should probably just `useCache`